### PR TITLE
preliminary realm changes

### PR DESCRIPTION
**Pull request by Daniela510**
put persistent state into a separate schema
separate data model from services/methods

### CODE REVIEW

File: lib/services/local_storage_service.dart [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        realm.all<LocalStorage>().first.persistentState = PersistentState();
+      });
+    }
+    if (realm.all<LocalUser>().isEmpty) {
+      realm.write(() {
+        realm.all<LocalStorage>().first.userData = LocalUser();
+      });
+    }
+  }
+  setWelcomeFlag(bool update) {
```
Comment (MaikelStuivenberg) [inline lib/services/local_storage_service.dart:33]: for consistency with the rest of the file, I would put "void" here as well (in front of the name).

File: lib/models/local_storage.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+@RealmModel()
+class _Donations {
+  // does it need to store more data?
+  // it SEEMS TO needs to store some id/ token
+  late final String mediumId;
+  late final double donationAmount;
+  // this should be an ISO8601 format
+  late final String dateTime;
```
Comment (AlexGivt) [inline lib/models/local_storage.dart:41]: should be a DateTime object? 
Best practise (in general): store date time in UTC and convert to local time in the UI

File: lib/models/local_storage.dart [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  String lastName = "";
+  String postalCode = "";
+  String country = "";
+  String password = "";
+  String appLanguage = "";
+  String timeZoneId = "";
+}
+
+@RealmModel()
+class _Donations {
```
Comment (AlexGivt) [inline lib/models/local_storage.dart:35]: single donation, so singular name?

File: lib/models/local_storage.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  // does it need to store more data?
+  // it SEEMS TO needs to store some id/ token
+  late final String mediumId;
+  late final double donationAmount;
+  // this should be an ISO8601 format
+  late final String dateTime;
+  late final String userId;
```
Comment (AlexGivt) [inline lib/models/local_storage.dart:42]: GUID/UUID

**Comment by phalox** [comment]
If this change unblocks the app testing, I would try to merge it as quickly as possible and then continue your work separately.

**Review by AlexGivt — APPROVED** [review]

**Pull request merged**